KmsClient.js

```diff
@@ -701,3 +743,36 @@ function _getRootZcapId({keystoreId, keyId}) {
   }
   return `${ZCAP_ROOT_PREFIX}${encodeURIComponent(suffix)}`;
 }
+
+/**
+ * Creates and signs the http headers for zCap invocation.
+ *
+ * @param {object} options - Options hashmap.
+ * @param {string} options.url - zCap target url.
+ * @param {string} options.method - HTTP method ('get', 'post').
+ * @param {object} [options.headers] - HTTP headers.
+ * @param {object} [options.operation] - Optional JSON payload (for POSTs).
+ * @param {object} [options.capability] - Optional existing capability.
+ * @param {{sign: Function}} options.invocationSigner - Key Signer object.
+ * @param {string} options.capabilityAction - zCap action to perform.
+ *
+ * @returns {Promise<object>} Results with the signed zcap headers object.
+ */
+async function _signCapabilityInvocation({
+  url, method = 'post', headers = this.defaultHeaders, operation, capability,
+  invocationSigner, capabilityAction
+}) {
+  try {
+    // sign HTTP header
+    return signCapabilityInvocation({
+      url, method: 'post', headers, json: operation, capability,
+      invocationSigner, capabilityAction
+    });
+  } catch(e) {
+    _handleClientError({
```

Comment:
I feel like this is trying to correct for insufficient errors emitted by `signCapabilityInvocation` -- and if those were more informative, we wouldn't need to call this error helper here, rather, we could just let errors bubble up. I've filed this issue:

https://github.com/digitalbazaar/http-signature-zcap-invoke/issues/19

I think my preference is to fix up the errors there so we don't need anything here. Thoughts?

Reply:
@dlongley - yeah, you're right. It would be better to handle this in `signCapabilityInvocation`.

Reply:
Ok, the error handling was improved in `signCapabilityInvocation` v3.1.0, and I reverted the usage of the helper method (now just calls the function directly, as before).